### PR TITLE
Update nginx_proxy add-on to use Bashio

### LINKED ISSUE #731: Update nginx_proxy add-on to use Bashio
The add-ons now have [Bashio](https://github.com/hassio-addons/bashio) available, which makes it easier and nicer to work with and is easier to understand. The `nginx_proxy` add-on needs to be rewritten to use the Bashio functions.

In general, this only requires changes to the add-on [`run.sh`](https://github.com/home-assistant/hassio-addons/blob/master/nginx_proxy/data/run.sh) script. A good example of an add-on that has been migrated already, is [`deconz`](https://github.com/home-assistant/hassio-addons/blob/master/deconz/data/run.sh).

Please note, nothing has to be done to make Bashio available, it is already baked-in into the add-on base images.

### PR DESCRIPTION
**Pull request by itsamenathan**
Closes #731

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @itsamenathan, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;hassio-addons#784).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: nginx_proxy/data/run.sh [hunk 1]
```diff
@@ -1,51 +1,50 @@
...
 
-if [ ! -f "$SNAKEOIL_CERT" ]; then
-    echo "[INFO] Creating 'snakeoil' self-signed certificate..."
+if ! bashio::fs.file_exists "${SNAKEOIL_CERT}"; then
+    bashio::log.info "Creating 'snakeoil' self-signed certificate..."
     openssl req -x509 -nodes -days 3650 -newkey rsa:2048 -keyout $SNAKEOIL_KEY -out $SNAKEOIL_CERT -subj '/CN=localhost'
 fi
 
-if [ "$CLOUDFLARE" == "true" ]; then
+if bashio::var.true "${CLOUDFLARE}"; then
```
Comment (pvizeli) [inline nginx_proxy/data/run.sh]: `bashio::config.true 'cloudflare'`

File: nginx_proxy/data/run.sh [hunk 2]
```diff
@@ -60,13 +59,13 @@ sed -i "s/%%DOMAIN%%/$DOMAIN/g" /etc/nginx.conf
 sed -i "s/%%HSTS%%/$HSTS/g" /etc/nginx.conf
 
 # Allow customize configs from share
-if [ "$CUSTOMIZE_ACTIVE" == "true" ]; then
-    CUSTOMIZE_DEFAULT=$(jq --raw-output ".customize.default" $CONFIG_PATH)
+if bashio::var.true "${CUSTOMIZE_ACTIVE}"; then
```
Comment (pvizeli) [inline nginx_proxy/data/run.sh]: Look comment above

File: nginx_proxy/data/run.sh (hunk 1 above)
Comment (itsamenathan) [reply nginx_proxy/data/run.sh]: Thanks! Made the change.

**Pull request merged**